src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,686 @@
-        resultPtr->_ValidateSettings();
-
-        return *resultPtr;
-    }
-    catch (const SettingsException& ex)
+    // See member description of _userProfileCount for an explanation of this gsl::span trickery.
+    for (const auto& profile : gsl::make_span(userSettings.profiles).subspan(_userProfileCount))
     {
-        auto settings{ winrt::make_self<implementation::CascadiaSettings>() };
-        settings->_loadError = ex.Error();
-        return *settings;
+        if (generatedProfileIds.emplace(profile->Guid()).second)
+        {
+            newGeneratedProfiles = true;
+        }
+        else
+        {
+            profile->Deleted(true);
+            profile->Hidden(true);
+        }
     }
-    catch (const SettingsTypedDeserializationException& e)
+
+    if (newGeneratedProfiles)
     {
-        auto settings{ winrt::make_self<implementation::CascadiaSettings>() };
-        std::string_view what{ e.what() };
-        settings->_deserializationErrorMessage = til::u8u16(what);
-        return *settings;
+        state->GeneratedProfiles(generatedProfileIds);
     }
 }
 
-// Function Description:
-// - Creates a new CascadiaSettings object initialized with settings from the
-//   hardcoded defaults.json.
-// Arguments:
-// - <none>
-// Return Value:
-// - a unique_ptr to a CascadiaSettings with the settings from defaults.json
-winrt::Microsoft::Terminal::Settings::Model::CascadiaSettings CascadiaSettings::LoadDefaults()
+// Call this method before passing SettingsLoader to the CascadiaSettings constructor.
+// It layers all remaining objects onto each other (those that aren't covered by LayerInboxOntoUser/LayerFragmentsOntoUser).
+
+void SettingsLoader::FinalizeLayering()
 {
-    auto resultPtr{ winrt::make_self<CascadiaSettings>() };
-
-    // We already have the defaults in memory, because we stamp them into a
-    // header as part of the build process. We don't need to bother with reading
-    // them from a file (and the potential that could fail)
-    resultPtr->_ParseJsonString(DefaultJson, true);
-    resultPtr->LayerJson(resultPtr->_defaultSettings);
-    resultPtr->_ResolveDefaultProfile();
-    resultPtr->_UpdateActiveProfiles();
-
-    // tag these profiles as in-box
-    for (const auto& profile : resultPtr->AllProfiles())
+    // Layer default globals -> user globals
+    userSettings.globals->InsertParent(inboxSettings.globals);
+    userSettings.globals->_FinalizeInheritance();
+    // Layer default profile defaults -> user profile defaults
+    userSettings.baseLayerProfile->InsertParent(inboxSettings.baseLayerProfile);
+    userSettings.baseLayerProfile->_FinalizeInheritance();
+    // Layer user profile defaults -> user profiles
+    for (const auto& profile : userSettings.profiles)
     {
-        const auto profileImpl{ winrt::get_self<implementation::Profile>(profile) };
-        profileImpl->Origin(OriginTag::InBox);
+        profile->InsertParent(0, userSettings.baseLayerProfile);

```

Comment:
You may want to include a note here explaining that using index=0 is what ensures that the user's `defaults` go on top of fragments and inbox and generated profiles.

Reply:
I've got a pent up change that replaces `InsertParent` entirely with `AddMost/LeastImportantParent`, which I'm planning to submit right after this PR. Even if these functions are more verbose, they are much more effective at conveying what is being done without writing comments.
As such I'd like to refrain from adding any such changes momentarily.